server/cmd/dcrdex/config.go

```diff
@@ -0,0 +1,564 @@
+	DBUser           string
+	DBPass           string
+	DBHost           string
+	DBPort           uint16
+	MarketsConfPath  string
+	RegFeeXPub       string
+	RegFeeConfirms   int64
+	RegFeeAmount     uint64
+	DEXPrivKey       *secp256k1.PrivateKey
+	RPCCert          string
+	RPCKey           string
+	RPCListen        []string
+	BroadcastTimeout time.Duration
+	AltDNSNames      []string
+	DefaultLogLevel  slog.Level
+	LogLevels        map[string]slog.Level
+}
+
+type flagsData struct {
+	// General application behavior
+	AppDataDir  string `short:"A" long:"appdata" description:"Path to application home directory"`
+	ConfigFile  string `short:"C" long:"configfile" description:"Path to configuration file"`
+	DataDir     string `short:"b" long:"datadir" description:"Directory to store data"`
+	LogDir      string `long:"logdir" description:"Directory to log output."`
+	DebugLevel  string `short:"d" long:"debuglevel" description:"Logging level {trace, debug, info, warn, error, critical}"`
+	MaxLogZips  int    `long:"maxlogzips" description:"The number of zipped log files created by the log rotator to be retained. Setting to 0 will keep all."`
+	ShowVersion bool   `short:"V" long:"version" description:"Display version information and exit"`
+
+	Testnet bool `long:"testnet" description:"Use the test network (default mainnet)"`
+	Simnet  bool `long:"simnet" description:"Use the simulation test network (default mainnet)"`
+
+	RPCCert     string   `long:"rpccert" description:"RPC server TLS certificate file"`
+	RPCKey      string   `long:"rpckey" description:"RPC server TLS private key file"`
+	RPCListen   []string `long:"rpclisten" description:"IP addresses on which the RPC server should listen for incoming connections"`
+	AltDNSNames []string `long:"altdnsnames" description:"A list of hostnames to include in the RPC certificate (X509v3 Subject Alternative Name)"`
+
+	MarketsConfPath  string        `long:"marketsconfpath" description:"Path to the markets configuration JSON file."`
+	BroadcastTimeout time.Duration `long:"bcasttimeout" description:"How long clients have to broadcast expected swap transactions following new blocks"`
+	RegFeeXPub       string        `long:"regfeexpub" description:"The extended public key for deriving Decred addresses to which DEX registration fees should be paid."`
+	RegFeeConfirms   int64         `long:"regfeeconfirms" description:"The number of confirmations required to consider a registration fee paid."`
+	RegFeeAmount     uint64        `long:"regfeeamount" description:"The registration fee amount in atoms."`
+	DEXPrivKeyPath   string        `long:"dexprivkeypath" description:"The path to a file containing the DEX private key for message signing."`
+
+	HTTPProfile  bool   `long:"httpprof" short:"p" description:"Start HTTP profiler."`
+	HTTPProfPath string `long:"httpprofprefix" description:"URL path prefix for the HTTP profiler."`
+	CPUProfile   string `long:"cpuprofile" description:"File for CPU profiling."`
```

Comment:
Description of default behavior would be helpful, e.g. port 9232

Also, when I try to set `httpprofprefix`, I get some kind of recursion issue.

Reply:
Ah, this doesn't work with `http.DefaultServeMux`.  I'm removing `HTTPProfPath` since it was of limited use.